streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -851,11 +853,19 @@ void runOnce() {
 
             do {
                 for (int i = 0; i < numIterations; i++) {
+                    advanceNowAndComputeLatency();
                     processed = taskManager.process(now);
 
                     if (processed > 0) {
+                        // It makes no difference to the outcome of these metrics when we record "0",
+                        // so we can just avoid the method call when we didn't process anything.
+                        processRateSensor.record(processed, now);
```

Comment:
I moved this into the block here because it makes no difference to the result, and it's a little more efficient.